### PR TITLE
Allow a standalone llpc build

### PR DESCRIPTION
**Pull request by trenouf**
Change-Id: Ifd26b5293fde7fc4bdb831021f7c81243b53b530

### CODE REVIEW

**Comment by trenouf** [comment]
This needs https://github.com/GPUOpen-Drivers/spvgen/pull/17 for an llpc standalone build to work.

File: cmake/CompilerStandalone.cmake [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    target_compile_definitions(khronos_vulkan_interface INTERFACE EXTERNAL_VULKAN_HEADERS=1)
+endif()
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../xgl)
+    target_include_directories(khronos_vulkan_interface INTERFACE ../xgl/icd/api/include/khronos)
+else()
+    target_include_directories(khronos_vulkan_interface INTERFACE ../../../icd/api/include/khronos)
+endif()
+
+set(LLPC_CLIENT_INTERFACE_MAJOR_VERSION "LLPC_INTERFACE_MAJOR_VERSION")
+set(PAL_CLIENT_INTERFACE_MAJOR_VERSION 9999)
```
Comment (kuhar) [inline cmake/CompilerStandalone.cmake:74]: Can you add a comment explaining why you picked a large number for this constant and why we need it?

**Review by kuhar — COMMENTED** [review]
This seems like a very welcome change.

Could we add a new CI check for this? Both to make sure that this configuration continues to build, and to have up-to-date instructions on how to set up the build tree and compile it.

Also, would it be possible to statically link with spvgen? Having to distribute an extra .so and remember to set its path as a command link argument is a nuisance. Do you know much work would that be?

File: llpc/docs/amdllpc.md [hunk 2]
```diff
@@ -106,36 +106,45 @@ amdllpc -auto-layout-desc -gfxip=9.0.0 a.frag
 amdllpc -gfxip=8.0.3 -o=c.elf b.pipe
 ```
 
-## Test with SHADERDB
-You can use [shaderdb](https://github.com/GPUOpen-Drivers/llpc/tree/master/test) to test llpc with standalone compiler and [spvgen](https://github.com/GPUOpen-Drivers/spvgen):
+## Driver build and lit testing
 
-By integrating with [lit](http://llvm.org/docs/CommandGuide/lit.html), the test will check the correctness of shader compilation result with the pattern specified in test files.
+LLPC is normally built as part of the [AMD Open Source Driver for Vulkan](../AMDVLK/README.md).
```
Comment (Flakebi) [inline llpc/docs/amdllpc.md:111]: Nice documentation!
I guess this could be an absolute link to the amdvlk repo?

**Comment by trenouf** [comment]
> Could we add a new CI check for this? Both to make sure that this configuration continues to build, and to have up-to-date instructions on how to set up the build tree and compile it.

Good idea.

> Also, would it be possible to statically link with spvgen? Having to distribute an extra .so and remember to set its path as a command link argument is a nuisance. Do you know much work would that be?

Sounds like a fair bit of work, mostly silly fiddly stuff.

I have a vague recollection that I did that once, but then discarded it because at the time we had a binary for spvgen in our internal source repository, and building it as a .a made it massive because of not-yet-commoned-up debug info.

**Review by amdrexu — APPROVED** [review]

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit e103752b4947bed7d9a3628e729cc37003a23331
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 56f2e165b889bd7ca147ff853e6f1372ed90d4fe</li>
<li>SPVGEN: dev 4846370016de9775b7518b51c08260387a7367c4</li>
<li>XGL: dev 7afa4bc3edba43cd54fc07be78f54f9e334ea524</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 59efb265ed3fd48137fed3282b7782742b88b2e1</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221339)</summary><ul><li>Built with version 1.2.5.1</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30317/55472 (54.7%)</li>
<li>Failed: 0/55472 (0.0%)</li>
<li>Not Supported: 25155/55472 (45.3%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30317/55472 (54.7%)</li>
<li>Failed: 0/55472 (0.0%)</li>
<li>Not Supported: 25155/55472 (45.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30067/54923 (54.7%)</li>
<li>Failed: 0/54923 (0.0%)</li>
<li>Not Supported: 24856/54923 (45.3%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 33889/55472 (61.1%)</li>
<li>Failed: 0/55472 (0.0%)</li>
<li>Not Supported: 21583/55472 (38.9%)</li>
</ul></details>


</ul></details>

**Pull request merged**